test/coverage/datamodel/UUriSerializerTest.cpp

```diff
@@ -23,16 +25,135 @@ class TestFixture : public testing::Test {
+	const std::string expectedAuthority = "192.168.1.10";
+	const uint32_t expectedUEID = 0x1100;
+	const uint32_t expectedMajorVersion = 0xFF;
+	const uint32_t expectedResourceID = 0x8000;
+
+	const UUri uri = AsString::deserialize(uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializer, DeSerializeEmptyUUriString) {
+	const std::string uuriAsString = "";
+	const std::string expectedAuthority = "";
+	const uint32_t expectedUEID = 0x0;
+	const uint32_t expectedMajorVersion = 0x0;
+	const uint32_t expectedResourceID = 0x0;
+
+	const UUri uri = AsString::deserialize(uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializer, DeSerializeUUriStringWithNoAuthority) {
+	const std::string uuriAsString = "/1100/FF/8000";
+	const std::string expectedAuthority = "";
+	const uint32_t expectedUEID = 0x1100;
+	const uint32_t expectedMajorVersion = 0xFF;
+	const uint32_t expectedResourceID = 0x8000;
+
+	const UUri uri = AsString::deserialize(uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializer, DeSerializeUUriStringWithInvalidNumberOfArgument) {
+	// Provided 5 arguments instead of 4 when authority exist
+	std::string uuriAsString = "//192.168.1.10/1100/FF/FF/8000";
+
+	ASSERT_THROW(AsString::deserialize(uuriAsString), std::runtime_error);
+
+	// UE ID is missing. Provided 3 arguments instead of 4 when authority exist.
+	uuriAsString = "//192.168.1.10/FF/8000";
+	ASSERT_THROW(AsString::deserialize(uuriAsString), std::runtime_error);
+
+	// Provided 4 arguments instead of 3 when authority does not exist.
+	uuriAsString = "/1100/FF/FF/8000";
+	ASSERT_THROW(AsString::deserialize(uuriAsString), std::runtime_error);
+
+	// UE ID is missing. Provided 2 arguments instead of 3 when authority exist.
+	uuriAsString = "/FF/8000";
+	ASSERT_THROW(AsString::deserialize(uuriAsString), std::runtime_error);
+}
+
+TEST_F(TestUUriSerializer, DeSerializeUUriStringWithInvalidArgument) {
+	// UE ID provided is invalid. It should be hex numeric
+	std::string uuriAsString = "//192.168.1.10/testUE/FF/8000";
+	ASSERT_THROW(AsString::deserialize(uuriAsString), std::invalid_argument);
+
+	// Major Version provided is invalid. It should be hex numeric
+	uuriAsString = "//192.168.1.10/1100/^%/8000";
+	ASSERT_THROW(AsString::deserialize(uuriAsString), std::invalid_argument);
+
+	// Resource ID provided is invalid. It should be hex numeric
+	uuriAsString = "//192.168.1.10/1100/FF/xyz";
```

Comment:
Add a test for an otherwise valid uri with no leading `/` and another with three (`///`). And one test with a trailing `/`

Reply:
Done. Added additional tests with no leading / and another one with leading /// and one with additional trailing /. All these tests should throw invalid argument